### ISSUE TITLE
Clustering with deep learning: Taxonomy and new methods

### ISSUE THREAD

**Issue by standing-o** [standing-o/Machine-learning_Paper_review#18]
- Authors : Aljalbout, Elie and Golkov, Vladimir and Siddiqui, Yawar and Strobel, Maximilian and Cremers, Daniel
- Journal : arXiv
- Year : 2018
- Link : https://arxiv.org/pdf/1801.07648.pdf

### Abstract
- Clustering methods based on deep neural networks ➔ High representational power

### Introduction
- The main objective of clustering is to separate data into groups of similar data points.
- The performance of current clustering methods is however highly dependent on the input data. 
- Dimensionality reduction and representation learning have been extensively used alongside clustering in order to map the input data into a feature space where separation is easier. By utilizing deep neural networks, it is possible to learn non-linear mappings that allow transforming data into more clustering-friendly representations without manual feature extraction/selection.
- The general pipeline of most deep-learning-based clustering methods
  <img src="https://user-images.githubusercontent.com/57218700/228452177-cecb224a-185f-4f3b-ac19-e63b53e37505.png" width=100%>
  - Auto-encoder is trained with the standard mean squared error reconstruction loss. The auto-encoder fine-tuned with a combined loss function consisting of the auto-encoder reconstruction loss and a clustering-specific loss.

#### Multi-layer Perceptron (MLP)
- Feedforward network. The output of every hidden layer is the input to next one
#### Convolutional Neural network (CNN)
- Useful for applications to images, if locality and shift-equivariance/invariance of feature extraction is desired.
#### Deep belief network (DBN)
- Generative graphical model, consisting of several layers of latent variable.
- Composed of several shallow networks such as restricted Boltzmann machines, s.t. the hidden layer of each sub-network serves as the visible layer of the next sub-network.
#### Generative Adversarial network (GAN)
- The generator learns a distribution of interest to produce samples. The discriminator learns to distinguish between real samples and generated ones.
#### Variational Auto encoder (VAE)
- A Bayesian network with an autoencoder architecture that learns the data distribution (generative model).

#### No Non-clustering loss
- Non additional non-clustering loss function is used and the network model is only constrained by the clustering loss.
- Danger of worse representations/results, or theoretically even collapsing clusters

#### Auto-encoder reconstruction loss
<img src="https://latex.codecogs.com/svg.latex?\Large&space;L=d_{\text{AE}}(x_i,f(x_i))=\Sigma_i||x_i-f(x_i)||^2" width=60%>

- During training, the decoder tries to reconstruct input x from representation z in a latent space Z, making sure that useful information has not been lost by the encoding phase.
- Auto-encoders can successfully learn useful representations in the cases where the output's dimensionality is different from the input's or when random noise is injected to the input.

#### Self-augmentation loss
<img src="https://latex.codecogs.com/svg.latex?\Large&space;L=-\frac{1}{N}\Sigma_{N}s(f(x),f(T(x)))" width=40%>

- T is the augmentation function, f(x) is the representation generated by the model and s is some measure of similarity.
- A loss term pushes together the representation of the original sample and their augmentations.

#### No Clustering loss
- Even if a neural network has only non-clustering losses, the features it extracts can be used for clustering after training.
